main/client/src/mill/main/client/FileToStreamTailer.java

```diff
@@ -0,0 +1,97 @@
+package mill.main.client;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Optional;
+
+public class FileToStreamTailer extends Thread implements AutoCloseable {
+
+	private final File file;
+	private final PrintStream stream;
+	private final int intervalMsec;
+
+	private Optional<BufferedReader> reader = Optional.empty();
+	// if true, we won't read the whole file, but only new lines
+	private boolean ignoreHead = true;
```

Comment:
Could we specify that the server creates a new log file for every run? That way we do not need to worry about ignoring anything, and can always read the log files starting from the top

Reply:
Not exactly. Even if we create new logs for each server run, as the log files are read by the client, each subsequent client would output all older logs.